src/packet_macros.rs

```diff
@@ -219,3 +237,18 @@ macro_rules! write_packet_field {
         $stream.write_chunk_section($value)
     };
 }
+
+macro_rules! translate_packet_field {
+    ($value:expr, $transdata:expr, Untranslated) => {
+        $value
+    };
+    ($value:expr, $transdata:expr, EntityId) => {
+        $value + ($transdata.entity_id_block * ENTITY_ID_BLOCK_SIZE)
+    };
+    ($value:expr, $transdata:expr, XChunk) => {
+        $transdata.map.x_origin
+    };
+    ($value:expr, $transdata:expr, XEntity) => {
+        $value + ($transdata.map.x_origin * CHUNK_SIZE) as f64
```

Comment:
Chunk and Entity x are different measures, chunk x is 16 times larger, so these need to be different.

Right now we only support a single section pillar per server, so we can just use the map origin and don't need to include the value for the chunk x translation. Eventually that'll need to change